### PR TITLE
define CELERY_QUEUES

### PR DESCRIPTION
**Pull request by eviljeff**
This may or may not fix the duplicate celery task issue

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
re-reading http://docs.celeryproject.org/en/3.1/userguide/routing.html#changing-the-name-of-the-default-queue I actually wonder if we ever have the default queue configured correctly… `CELERY_DEFAULT_QUEUE` is only being used as a reference in that example. But http://docs.celeryproject.org/en/3.1/configuration.html#celery-default-queue says clearly "If CELERY_QUEUES is not specified then it is automatically created containing one queue entry, where this name is used as the name of that queue." which makes it even worse given that we have various queues defined in our `CELERY_ROUTES` configuration.

Let's maybe define **everything** explicitly in `CELERY_QUEUES` too? I have the feeling we're sitting on a huge time bomb somehow 🚀

**Comment by eviljeff** [comment]
yeah... we're setting a `_DEFAULT_QUEUE` without defining that `Queue` (or any `Queue`s).

**Comment by diox** [comment]
But... isn't it clearly working correctly atm ? I mean, we do have different queues set up (including one called "default"), and tasks do get distributed correctly in them, right ? I'm in favor of declaring the queues explicitly but I'd like to understand what's going on instead of blindly committing stuff.

**Comment by eviljeff** [comment]
We define `CELERY_ROUTES` but no `CELERY_QUEUES` so Celery creates the new queues automatically (as `CELERY_CREATE_MISSING_QUEUES` is on by default).  So we're specify a `CELERY_DEFAULT_QUEUE` that doesn't exist - my guess is somewhere there is a bug that both automatically creates a `default` queue based on our setting, and uses the default-default queue `celery` because it doesn't exist.

**Comment by diox** [comment]
Is the `celery` queue guaranteed to exist though ? We don't define it in `CELERY_ROUTES` either do we ?

**Comment by eviljeff** [comment]
we don't reference `celery` in `CELERY_ROUTES`, no.  But we don't _define_ any queues at the moment - Celery just creates them when we reference them and they don't exist.  The most correct way is probably as @EnTeQuAk said - we should define all the queues we reference in `CELERY_ROUTES` and elsewhere in `CELERY_QUEUES`.

**Comment by diox** [comment]
My point was: why change `CELERY_DEFAULT_QUEUE` from `default` to `celery` if `celery` isn't defined any more than `default` is ? What does that give us ? I do agree we should define `CELERY_QUEUES`, I just don't see the point of changing `CELERY_DEFAULT_QUEUE`.

**Comment by eviljeff** [comment]
If its a bug like I think, then adding `default` to `CELERY_QUEUES` (along with some others) should fix it (instead of changing `CELERY_DEFAULT_QUEUE` like this pull does)

**Comment by EnTeQuAk** [comment]
> If its a bug like I think, then adding default to CELERY_QUEUES (along with some others) should fix it (instead of changing CELERY_DEFAULT_QUEUE like this pull does)

I'd be 👍 on defining CELERY_QUEUES with all our queues that we have. Explicit is always better than implicit

**Comment by eviljeff** [comment]
https://github.com/mozilla/addons-server/pull/4946/commits/fac5d95e105a773b411de8f8e7bfa9bdef4c462c adds all the queues in `CELERY_ROUTES` though I'm beginning to think this is a little silly - unless we're treating some of the queues differently then we don't need so many.   And if they're going to default we don't even need them in `CELERY_ROUTES`

**Comment by diox** [comment]
I agree. We should double-check the dev/stage/prod configuration with ops, making sure all queues exists right now and if some have special configuration.

**Comment by eviljeff** [comment]
ref: https://github.com/mozilla-services/cloudops-deployment/blob/master/projects/addons/puppet/modules/olympia/manifests/celery.pp#L30:L62 for queue configurations

File: src/olympia/lib/settings_base.py [hunk 1]
```diff
@@ -1057,6 +1059,27 @@ def JINJA_CONFIG():
     'olympia.lib.es.management.commands.reindex',
 )
 
+CELERY_QUEUES = (
```
Comment (diox) [inline src/olympia/lib/settings_base.py]: Can you add a test that all queues used in `CELERY_ROUTES` + `CELERY_DEFAULT_QUEUE` are defined in `CELERY_QUEUES` ?

Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: https://github.com/mozilla/addons-server/pull/4946/commits/9fe269603c08f9694595e3871d5af2194f147356

File: src/olympia/lib/settings_base.py [hunk 2]
```diff
@@ -1121,9 +1144,6 @@ def JINJA_CONFIG():
         'queue': 'addons'},
     'olympia.addons.tasks.version_changed': {'queue': 'addons'},
 
-    # Files (goes to devhub queue).
-    'olympia.files.tasks.update_webext_descriptions_all': {'queue': 'devhub'},
```
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: We need this to be missing from `CELERY_ROUTES` to check that setting `CELERY_QUEUES` fixes the duplicate task problem... but at the same time it should (ideally) be defined.

Comment (EnTeQuAk) [inline src/olympia/lib/settings_base.py]: Can we have that change in a separate commit so that we can easily rebase it later after testing?

**Comment by diox** [comment]
Let's not merge it today, wait for next week's tag.

File: src/olympia/lib/settings_base.py (hunk 2 above)
Comment (eviljeff) [inline src/olympia/lib/settings_base.py]: rebased into two commits; the latter that only removes a queue from `CELERY_ROUTES` to test

**Comment by eviljeff** [comment]
redid as https://github.com/mozilla/addons-server/pull/4996 🐋

**Pull request merged**